airflow/api_connexion/openapi/v1.yaml

```
@@ -2288,7 +2288,7 @@ paths:
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/Role'
+                $ref: '#/components/schemas/User'
```

```suggestion
                $ref: '#/components/schemas/UserCollectionItem'
```
UserCollectionItem is more appropriate here. You also need to add test to protect accidental changing in the future

---

Done, thanks for the suggestion. Regarding the tests, it seems there are already some tests [here](https://github.com/apache/airflow/blob/main/tests/api_connexion/endpoints/test_user_endpoint.py#L529) but regardless of the schema referenced in `v1.yaml`, it always return the same schema of data (the correct one). Any idea why?

---

Any idea @ephraimbuddy?

---

It's strange to me that the response is not enforced in the spec. I had thought that the spec provides input and output validation but this proves otherwise